src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -1368,16 +1368,22 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		});
 
 		await this._extensionService.activateByEvent(`onConnect:${connection.providerName}`);
-
+		if (this._providers.get(connection.providerName) === undefined) {
+			await this.handleUnsupportedProvider(connection.providerName);
+			throw new Error(nls.localize('connection.providerNotFound', "Connection provider '{0}' not found", connection.providerName));
```

Comment:
Missed the original PR, but why are you throwing? Shouldn't the extension be installed after calling handleUnsupportedProvider? So you should be able to continue this logic instead of making the user click it again.

Reply:
Installing the provider refreshes the tree and replaces the node. I don't think we can still use the same expansion call to complete the expansion request.

Reply:
Hrm, that's unfortunate. But I doubt that's an easy fix so we'll leave that as a future enhancement if needed (but given that it's essentially a one-time thing it's probably fine as is)

I would suggest adding a comment in main to this code saying that though so it's clear why we don't/can't (don't need to port it here)